apps/remix-ide/src/app/tabs/compileTab/compileTab.js

```diff
@@ -80,6 +80,22 @@ class CompileTab {
 
   runCompiler () {
     try {
+      console.log('mode is - ', this.fileManager.mode)
+      if (this.fileManager.mode === 'localhost') {
+        console.log('calling compilehardhat')
+        const { currentVersion, optimize, runs } = this.compiler.state
+        const fileContent = `module.exports = {
+          solidity: '${currentVersion.substring(0, currentVersion.indexOf('+commit'))}',
+          settings: {
+            optimizer: {
+              enabled: ${optimize},
+              runs: ${runs}
+            }
+          }
+        }
+        `
+        this.fileManager.compileWithHardhat(fileContent).then(console.log)
```

Comment:
could you directly do the plugin call from the solidity compiler?

Reply:
ok

Reply:
that'd be still better to not have the hardhat plugin call in the filemanager. 
> have to make this method async, even for workspaces apart from local

`compileWithHardhat` is also `async` .. I don't see where is the issue.
And you don't need to `await` the hardhat compilation, just to manage the result in a promise resolve/reject

Reply:
this is done